utils/src/wizard/AzureWizard.ts

```diff
@@ -172,12 +174,40 @@ export class AzureWizard<T extends IInternalActionContext> implements types.Azur
                     await step.execute(this._context, internalProgress);
                     currentStep += 1;
                 }
-
                 step = steps.pop();
             }
         });
     }
 
+    private async withProgress(executeOptions: types.AzureWizardExecuteOptions, options: vscode.ProgressOptions, task: (progress: vscode.Progress<{ message?: string; increment?: number }>, token: vscode.CancellationToken) => Promise<void>): Promise<void> {
+        if (executeOptions.activity) {
+            const activity = new ExecuteActivity({
+                title: executeOptions.activity.name ?? this.title ?? 'Azure operation',
+                context: this._context,
+            }, async (activityProgress) => {
+
+                await vscode.window.withProgress(options, async (progress: vscode.Progress<{ message?: string; increment?: number }>, token: vscode.CancellationToken): Promise<void> => {
```

Comment:
Should activity log or progress toast be an either/or thing?

Reply:
I think we should make it a setting with possible values: "Activity", "Notification", "Both", "None"

Reply:
Good idea!

Reply:
Should I leave adding this setting for a later PR?

Reply:
Up to you, I think it fits close enough that it'd be fine to include here.